tests/integration.rs

```diff
@@ -781,6 +792,59 @@ fn input_focus() {
     f.wm_delete_window(&mut connection, win1, surface1);
 }
 
+#[test]
+fn activation_x11_to_x11() {
+    let mut f = Fixture::new();
+    let mut connection = Connection::new(&f.display);
+
+    let window1 = connection.new_window(connection.root, 0, 0, 20, 20, false);
+    let surface1 = f.map_as_toplevel(&mut connection, window1);
+    let window2 = connection.new_window(connection.root, 0, 0, 20, 20, false);
+    let surface2 = f.map_as_toplevel(&mut connection, window2);
+
+    f.testwl.focus_toplevel(surface2);
+    {
+        let focus = connection.get_reply(&x::GetInputFocus {}).focus();
+        assert_eq!(window2, focus);
+
+        let reply = connection.get_reply(&x::GetProperty {
+            delete: false,
+            window: connection.root,
+            property: connection.atoms.net_active_window,
+            r#type: x::ATOM_WINDOW,
+            long_offset: 0,
+            long_length: 1,
+        });
+
+        assert_eq!(&[window2], reply.value::<x::Window>());
+    };
+
+    connection.send_client_message(&x::ClientMessageEvent::new(
```

Comment:
```suggestion
    std::thread::sleep(Duration::from_millis(1));
    connection.send_client_message(&x::ClientMessageEvent::new(
```
For some reason, adding this mini sleep right before sending the client message enables it to actually get processed. Why? Don't know, maybe a race condition. Don't care too much.